titus-api/src/main/java/com/netflix/titus/api/jobmanager/model/job/Volume.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2018 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.titus.api.jobmanager.model.job;
+
+import javax.validation.Valid;
+
+public class Volume {
+
+    @Valid
+    private final String name;
+
+    private final SharedContainerVolumeSource sharedContainerVolumeSource;
```

Comment:
Is this value relevant for all volume types?

Reply:
It is not, but I'm not sure how to make it optional except to just set it to `null` when not related.

Do you think this would be better implemented as.. inheritance?

Reply:
Yes, we use inheritance in such cases. Please, have a look at the disruption budget model.

Reply:
ty. I've done this, please give this extra scrutiny because this is my first time interacting with inheritance in java.